pydantic_ai_slim/pydantic_ai/models/bedrock.py

```diff
@@ -0,0 +1,445 @@
+    _timestamp: datetime = field(default_factory=_utils.now_utc)
+
+    async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
+        """Return an async iterator of [`ModelResponseStreamEvent`][pydantic_ai.messages.ModelResponseStreamEvent]s.
+
+        This method should be implemented by subclasses to translate the vendor-specific stream of events into
+        pydantic_ai-format events.
+        """
+        chunk: ConverseStreamOutputTypeDef
+        async for chunk in AsyncIteratorWrapper(self._event_stream):
+            # TODO: Switch this to `match` when we drop Python 3.9 support
+            if 'messageStart' in chunk:
+                continue
+            if 'messageStop' in chunk:
+                continue
+            if 'metadata' in chunk:
+                if 'usage' in chunk['metadata']:
+                    self._usage += self._map_usage(chunk['metadata'])
+                continue
+            if 'contentBlockStart' in chunk:
+                index = chunk['contentBlockStart']['contentBlockIndex']
+                start = chunk['contentBlockStart']['start']
+                if 'toolUse' in start:
+                    tool_use_start = start['toolUse']
+                    tool_id = tool_use_start['toolUseId']
+                    tool_name = tool_use_start['name']
+                    yield self._parts_manager.handle_tool_call_delta(
+                        vendor_part_id=index,
+                        tool_name=tool_name,
+                        args=None,
+                        tool_call_id=tool_id,
+                    )
+
+            if 'contentBlockDelta' in chunk:
+                index = chunk['contentBlockDelta']['contentBlockIndex']
+                delta = chunk['contentBlockDelta']['delta']
+                if 'text' in delta:
+                    yield self._parts_manager.handle_text_delta(vendor_part_id=index, content=delta['text'])
+                if 'toolUse' in delta:
+                    tool_use = delta['toolUse']
+                    yield self._parts_manager.handle_tool_call_delta(
+                        vendor_part_id=index,
+                        tool_name=tool_use.get('name'),
+                        args=tool_use.get('input'),
+                        tool_call_id=tool_id,
+                    )
+
+    @property
+    def timestamp(self) -> datetime:
+        return self._timestamp
+
+    @property
+    def model_name(self) -> str:
+        """Get the model name of the response."""
+        return self._model_name
+
+    def _map_usage(self, metadata: ConverseStreamMetadataEventTypeDef) -> result.Usage:
+        return result.Usage(
+            request_tokens=metadata['usage']['inputTokens'],
+            response_tokens=metadata['usage']['outputTokens'],
+            total_tokens=metadata['usage']['totalTokens'],
+        )
+
+
+# TODO(Marcelo): Should we call this `BedrockConverseAPI` instead?
+@dataclass(init=False)
+class BedrockModel(Model):
```

Comment:
@dmontagu Bedrock is not a model, it's a provider... Converse API is the API/interface.

How should we call this?